packages/generator/templates/app/app/auth/mutations/resetPassword.ts

```diff
@@ -0,0 +1,51 @@
+import { SessionContext } from "blitz"
+import db from "db"
+import { ResetPasswordInput, ResetPasswordInputType } from "../validations"
+import { hashToken, hashPassword } from "../auth-utils"
+
+export class ResetPasswordError extends Error {
+  name = "ResetPasswordError"
+  message = "Reset password link is invalid or it has expired."
+}
+
+export default async function resetPassword(
+  input: ResetPasswordInputType,
+  _ctx: { session?: SessionContext } = {}
+) {
+  const { password, token } = ResetPasswordInput.parse(input)
+
+  // 1. Try to find this token in the database
+  // Have to first decodeURIComponent(token) because encodeURIComponent
+  // was used in forgotPassword.ts when sending to the browser
+  const hashedToken = hashToken(decodeURIComponent(token))
+  const possibleToken = await db.token.findOne({
+    where: { hashedToken, type: "RESET_PASSSWORD" },
```

Comment:
This line should be changed to account for the fact that there's a unique index on hashedToken and type:
`    where: { hashedToken_type: { hashedToken, type: 'RESET_PASSSWORD' } },`

Reply:
@aaronfulkerson I'm not understanding that change? The code in the PR is working without this change.

Reply:
![Screen Shot 2020-10-10 at 2 24 16 PM](https://user-images.githubusercontent.com/31112737/95664358-64920d00-0b04-11eb-84f0-c1083380006d.png)

That's strange. According to the typings and the way Prisma works, you can only pass unique fields to findOne.

Reply:
Oh, you're right! My production code is different than this. I think the real fix is to change `findOne` to `findFirst` and then the code in the PR should work.